test/functional/runner/index.js

```diff
@@ -0,0 +1,591 @@
+    return it.skip;
+  }
+
+  return it;
+}
+
+function generateTopologyTests(testSuites, testContext) {
+  testSuites.forEach(testSuite => {
+    const environmentRequirementList = parseRunOn(testSuite.runOn);
+
+    environmentRequirementList.forEach(requires => {
+      const suiteName = `${testSuite.name} - ${requires.topology.join()}`;
+      describe(suiteName, {
+        metadata: { requires },
+        test: function() {
+          beforeEach(() => prepareDatabaseForSuite(testSuite, testContext));
+          afterEach(() => testContext.cleanupAfterSuite());
+
+          testSuite.tests.forEach(spec => {
+            const maybeSkipIt = shouldSkipTest(spec);
+            maybeSkipIt(spec.description, function() {
+              let testPromise = Promise.resolve();
+
+              if (spec.failPoint) {
+                testPromise = testPromise.then(() => testContext.enableFailPoint(spec.failPoint));
+              }
+
+              // run the actual test
+              testPromise = testPromise.then(() =>
+                runTestSuiteTest(this.configuration, spec, testContext)
+              );
+
+              if (spec.failPoint) {
+                testPromise = testPromise.then(() => testContext.disableFailPoint(spec.failPoint));
+              }
+
+              return testPromise.then(() => validateOutcome(spec, testContext));
+            });
+          });
+        }
+      });
+    });
+  });
+}
+
+// Test runner helpers
+function prepareDatabaseForSuite(suite, context) {
+  context.dbName = suite.database_name;
+  context.collectionName = suite.collection_name;
+
+  const db = context.sharedClient.db(context.dbName);
+  const coll = db.collection(context.collectionName);
+
+  return db
+    .admin()
+    .command({ killAllSessions: [] })
+    .catch(() => {}) // ignore any error from this
+    .then(() => coll.drop({ writeConcern: 'majority' }))
+    .catch(err => {
+      if (!err.message.match(/ns not found/)) throw err;
+    })
+    .then(() => {
+      if (suite.key_vault_data) {
+        const dataKeysCollection = context.sharedClient.db('admin').collection('datakeys');
+        return dataKeysCollection
+          .drop({ w: 'majority' })
+          .catch(err => {
+            if (!err.message.match(/ns not found/)) {
+              throw err;
+            }
+          })
+          .then(() => {
+            if (suite.key_vault_data.length) {
+              return dataKeysCollection.insertMany(suite.key_vault_data, { w: 'majority' });
+            }
+          });
+      }
+    })
+    .then(() => {
+      const options = { w: 'majority' };
+      if (suite.json_schema) {
+        options.validator = { $jsonSchema: suite.json_schema };
+      }
+
+      return db.createCollection(context.collectionName, options);
+    })
+    .then(() => {
+      if (suite.data && Array.isArray(suite.data) && suite.data.length > 0) {
+        return coll.insertMany(suite.data, { w: 'majority' });
```

Comment:
What is `coll` when we get here since we did `coll.drop()` earlier?

Reply:
mongodb will automatically create the collection if it didn't already exist